### DISCUSSION TITLE
SAPI?

### DISCUSSION THREAD

**Discussion by willwade** [coqui-ai/TTS#1609] [General Q&A]
We'd love to use this in our field of AAC. But is there a way/plans to make SAPI files/installers at all?!! (I have no idea if even possible)

**Comment by erogol** [comment]
What is SAPI?

**Reply by ghost** [reply]
It stands for Speech application programing interface and is used by Microsoft for speech in windows. Support for this plus the Mac OS speech manager would be great to have, however I don't know how easy it would be to implement.

**Reply by willwade** [reply]
That’s it. We work in AAC - alternative and augmentative communication. Think Stephen Hawking. We desperately need to create TTS voices for regional accents and new languages. Making a documented SDK to just use TTS in software - would be great. Using SAPI helps because a lot of software on windows use this.

**Reply by erogol** [reply]
I think it is beyond 🐸TTS. Maybe at some point, we can have a different project but 🐸TTS will be forcing on creating models.

**Reply by erogol** [reply]
How far off SAPI is supported by the commercial tools? Is it only Microsoft?

**Reply by willwade** [reply]
Well I’m no expert but yes. Each OS has their own synthesiser/sdk - some have got SAPI working on Linux but it’s hacky https://github.com/mjakal/sapi5_on_linux 

this is misleading but thought I’d dump it here https://en.m.wikipedia.org/wiki/Comparison_of_speech_synthesizers

**Reply by willwade** [reply]
Nb. Maybe relevant to put here for anybody else - I see the tts command line / python way of generating speech https://tts.readthedocs.io/en/latest/inference.html#on-the-commandline-tts but is there other ways for say swift or js systems?

**Reply by erogol** [reply]
we just have the ways mentioned in the doc

**Comment by BitBarrel** [comment]
I also would like to see a proper interface to generate speech from code on Windows. A C# interface as is available with coqui STT would be really helpful.

**Reply by willwade** [reply]
Yeah - basically we’d be super interested in creating more models for different accents and languages but for adoption in aac software we really could do with other interfaces that work across the plethora of software architecture out there.
